front/lib/models/workspace.ts

```diff
@@ -68,6 +77,16 @@ Workspace.init(
       type: DataTypes.BOOLEAN,
       defaultValue: false,
     },
+    whiteListedProviders: {
+      type: DataTypes.ARRAY(DataTypes.ENUM(...modelProviderEnumValues)),
+      defaultValue: null,
```

Comment:
To double check once you add the migration if it does set the default value properly.

Reply:
Indeed the migration doesn't set the default value nor the type properly. I modified the types to be strings and added a validator instead.

Reply:
Should we migrate to populate the DB and set it at workspace creation?

That way we can remove the |null from the type which will make things a bit simpler no?

Reply:
Do we want to though ? `null` means all providers. If non-null, then you only have the whitelist right ?

Reply:
so:
`null` -> all providers
`["x"]` -> only "x"
`[]` -> no provider (but should not be possible)

Reply:
I did it that way such that `null` would be "our" default provider while having a value would me the user's. 

Not strongly held on this as I think one option makes it a bit more handy at creation time while the other at manipulation time

Reply:
I think the null is nice for embedding provider too, since it means we don't have to change it for everyone in DB when we switch our default.

Reply:
unless we create some kind of magic string such as "DUST_DEFAULT_EMBEDDING_PROVIDER"

Reply:
Yeah it's all easy to tweak anyway

Agreed that there is value in potentially switching our default for all users that didn't modify it. But also not modifying it does not mean they didn't see it and might assume we would not change it...